gitlab/resource_gitlab_project_access_token_test.go

```diff
@@ -0,0 +1,204 @@
+	pat     *gitlab.ProjectAccessToken
+	project string
+	token   string
+}
+
+func testAccCheckGitlabProjectAccessTokenAttributes(patWrap *testAccGitlabProjectAccessTokenWrapper, want *testAccGitlabProjectAccessTokenExpectedAttributes) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		pat := patWrap.pat
+		if pat.Name != want.name {
+			return fmt.Errorf("got Name %q; want %q", pat.Name, want.name)
+		}
+
+		if pat.ExpiresAt.String() != want.expiresAt {
+			return fmt.Errorf("got ExpiresAt %q; want %q", pat.ExpiresAt.String(), want.expiresAt)
+		}
+
+		for _, scope := range pat.Scopes {
+			if !want.scopes[scope] {
+				return fmt.Errorf("got a not wanted Scope %q, received %v", scope, pat.Scopes)
+			}
+			want.scopes[scope] = false
+		}
+		for k, v := range want.scopes {
+			if v {
+				return fmt.Errorf("not got a wanted Scope %q, received %v", k, pat.Scopes)
+			}
+		}
+
+		git, err := gitlab.NewClient(patWrap.token, gitlab.WithBaseURL((testAccProvider.Meta().(*gitlab.Client)).BaseURL().String()))
+		if err != nil {
+			return fmt.Errorf("Cannot use the token to instantiate a new client %s", err)
+		}
+		_, _, err = git.ProjectMembers.ListAllProjectMembers(patWrap.project, nil)
+		if err != nil {
+			return fmt.Errorf("Cannot use the token to perform an API call %s", err)
+		}
+
+		return nil
+	}
+}
+
+func testAccCheckGitlabProjectAccessTokenDestroy(s *terraform.State) error {
+	conn := testAccProvider.Meta().(*gitlab.Client)
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "gitlab_project" {
+			continue
+		}
+
+		gotRepo, resp, err := conn.Projects.GetProject(rs.Primary.ID, nil)
+		if err == nil {
+			if gotRepo != nil && fmt.Sprintf("%d", gotRepo.ID) == rs.Primary.ID {
+				if gotRepo.MarkedForDeletionAt == nil {
+					return fmt.Errorf("Repository still exists")
+				}
+			}
+		}
+		if resp.StatusCode != 404 {
+			return err
+		}
+		return nil
+	}
+	return nil
+}
+
+func testAccGitlabProjectAccessTokenConfig(rInt int) string {
+	return fmt.Sprintf(`
+resource "gitlab_project" "foo" {
+  name = "foo-%d"
+  description = "Terraform acceptance tests"
+
+  # So that acceptance tests can be run in a gitlab organization
+  # with no billing
+  visibility_level = "public"
+}
+
+resource "gitlab_project_access_token" "bar" {
+	name = "my project token"
+    project = gitlab_project.foo.id
+    expires_at = "2022-04-01"
+    scopes = ["read_repository" , "api", "write_repository", "read_api"]
+}
+	`, rInt)
+}
+
+func testAccGitlabProjectAccessTokenUpdateConfig(rInt int) string {
+	return fmt.Sprintf(`
+resource "gitlab_project" "foo" {
+  name = "foo-%d"
+  description = "Terraform acceptance tests"
+
+  # So that acceptance tests can be run in a gitlab organization
+  # with no billing
+  visibility_level = "public"
+}
+
+resource "gitlab_project_access_token" "bar" {
+	name = "my new project token"
+    project = gitlab_project.foo.id
+    expires_at = "2022-05-01"
+    scopes = ["api"]
+}
+	`, rInt)
+}
```

Comment:
May be out of scope but I think it could be interesting to have a test case on the resource usage itself, e.g.: Adding the token to a CI/CD Variable and asserting it was correctly populated.

Other than that everything looks great :+1:

Reply:
Not sure I understand you suggestion here.

Reply:
Having a test that uses the created resource itself (the project token), for example, in a project variable.

In practice should be as simple as a new test case where the config would have the following as well:

```
resource "gitlab_project_variable" "sample_project_variable" {
    project = gitlab_project.foo.id
    key = "sample_variable_key"
    value = gitlab_project_access_token.bar.token
}
```

Maybe this could be added to  `## Example Usage` in docs/resources/project_access_token.md as well

Reply:
I implemented the suggested change in my last commit.

Reply:
Thank you, this is great!